packages/synapse-react-client/src/components/Plot/PlotsContainer.tsx

```diff
@@ -0,0 +1,323 @@
+        ...customPlots.map(
+          (plotProps): UiPlotState => ({
+            plotId: {
+              __custom: true,
+              title: plotProps.title ?? '',
+            },
+            isHidden: false,
+            plotType: convertPlotlyPlotTypeToFacetNavPlotType(plotProps.type),
+          }),
+        ),
+      )
+    }
+
+    if (facetNavPanelPropsArray && facetNavPanelPropsArray.length > 0) {
+      newPlotUiStateArray.push(
+        ...facetNavPanelPropsArray.map(facetPlotProps => ({
+          plotId: facetPlotProps.facetToPlot,
+          isHidden: false,
+          plotType: DEFAULT_PLOT_TYPE,
+        })),
+      )
+    }
+    setPlotUiStateArray(newPlotUiStateArray)
+  }, [customPlots, facetNavPanelPropsArray])
+
+  const lastQueryRequest = getCurrentQueryRequest()
+
+  // when 'show more/less' is clicked
+  const onShowMoreClick = (shouldShowMore: boolean) => {
+    setPlotUiStateArray(plotUiStateArray => {
+      return plotUiStateArray.map((item, index) => {
+        if (shouldShowMore) {
+          // show everything
+          return { ...item, isHidden: false }
+        }
+        // otherwise hide everything except the first three items
+        return { ...item, isHidden: index >= DEFAULT_VISIBLE_PLOTS }
+      })
+    })
+  }
+
+  // don't show hidden plots
+  const isPlotHiddenInGrid = (plotId: PlotIdentifier) => {
+    const itemHidden = plotUiStateArray.find(
+      item =>
+        plotMatchesDefinition(plotId, item.plotId) && item.isHidden === true,
+    )
+    const result = itemHidden !== undefined
+    return result
+  }
+
+  const showMoreButtonState = useMemo<ShowMoreState>(() => {
+    if (
+      // if at least one item is hidden
+      plotUiStateArray.find(item => item.isHidden === true)
+    ) {
+      return 'MORE'
+    } else if (plotUiStateArray.length <= DEFAULT_VISIBLE_PLOTS) {
+      return 'NONE'
+    } else {
+      return 'LESS'
+    }
+  }, [plotUiStateArray])
+
+  // hides facet graph
+  const hidePlotInGrid = (plotId: PlotIdentifier) => {
+    setUiPropertyForPlot(plotId, 'isHidden', true)
+  }
+
+  const setPlotType = (plotId: PlotIdentifier, plotType: PlotType) => {
+    setUiPropertyForPlot(plotId, 'plotType', plotType)
+  }
+
+  const setUiPropertyForPlot = (
+    plotId: PlotIdentifier,
+    propName: keyof UiPlotState,
+    value: boolean | PlotType, // 'the possible values of the above type' (currently can't be specified in TS using symbols)
+  ) => {
+    setPlotUiStateArray(plotUiStateArray =>
+      plotUiStateArray.map(item =>
+        plotMatchesDefinition(plotId, item.plotId)
+          ? { ...item, [propName]: value }
+          : item,
+      ),
+    )
+  }
+
+  const colorTracker: { facet: UniqueFacetIdentifier; colorIndex: number }[] =
+    // Todo: fix to work with customPlots
+    getFacets(data, facetsToPlot).map((el, index) => {
+      return {
+        facet: { columnName: el.columnName, jsonPath: el.jsonPath },
+        colorIndex: index,
+      }
+    })
```

Comment:
Need to figure out what this is doing, but I think it may be out of scope to actually get custom plots to use these colors